src/marshmallow/fields.py

```diff
@@ -242,21 +242,11 @@ def _validate(self, value):
         """Perform validation on ``value``. Raise a :exc:`ValidationError` if validation
         does not succeed.
         """
-        errors = []
-        kwargs = {}
-        for validator in self.validators:
-            try:
-                r = validator(value)
-                if not isinstance(validator, Validator) and r is False:
-                    raise self.make_error("validator_failed")
-            except ValidationError as err:
-                kwargs.update(err.kwargs)
-                if isinstance(err.messages, dict):
-                    errors.append(err.messages)
-                else:
-                    errors.extend(err.messages)
-        if errors:
-            raise ValidationError(errors, **kwargs)
+        self._validate_all(value)
+
+    @property
+    def _validate_all(self):
+        return And(self.validators, error=self.error_messages["validator_failed"])
```

Comment:
I initially imagined building this in the constructor and avoiding the errors list for basic validators, but I did not consider that the error object requires a list wrapper to keep things normalized. This won't provide the optimization I was hoping for, but I still think it is a worth while refactor.

Reply:
Yeah, I tried to build this in the constructor first, but it ended up being hairy to make it work with `Field`s that insert validators (e.g. Email, URL).

There still might be a clean way to do it, but I think we can leave that for the future.